src/main/java/com/google/googleidentity/oauth2/util/OAuth2Utils.java

```diff
@@ -47,10 +48,30 @@
 
     }
 
+    /**
+     * @return whether the uri matches one of the uri in uriList
+     */
+    public static boolean matchUri(List<String> uriList, String uri) {
```

Comment:
This method needs to correctly restrict client IDs to specific URLs. Make sure there are associated unit tests for each codepath here.